tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -222,144 +199,167 @@ Copyright (C) Aqua Security inc.
+#define UID_GREATER                     1
+#define PID_LESS                        2
+#define PID_GREATER                     3
+#define MNTNS_LESS                      4
+#define MNTNS_GREATER                   5
+#define PIDNS_LESS                      6
+#define PIDNS_GREATER                   7
 
-#define LESS_NOT_SET    0
-#define GREATER_NOT_SET ULLONG_MAX
+#define LESS_NOT_SET                    0
+#define GREATER_NOT_SET                 ULLONG_MAX
 
-#define DEV_NULL_STR    0
+#define DEV_NULL_STR                    0
 
-#define CONT_ID_LEN 12
-#define CONT_ID_MIN_FULL_LEN 64
+#define CONT_ID_LEN                     12
+#define CONT_ID_MIN_FULL_LEN            64
 
-#define CONTAINER_EXISTED  1  // container existed before tracee was started
-#define CONTAINER_CREATED  2  // new cgroup path created
-#define CONTAINER_STARTED  3  // a process in the cgroup executed a new binary
+#define CONTAINER_EXISTED               1         // container existed before tracee was started
+#define CONTAINER_CREATED               2         // new cgroup path created
+#define CONTAINER_STARTED               3         // a process in the cgroup executed a new binary
 
 #ifndef CORE
-#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 2, 0)
-// Use lower values on older kernels, where the instruction limit is 4096
-#define MAX_STR_ARR_ELEM      40
-#define MAX_ARGS_STR_ARR_ELEM 15
-#define MAX_PATH_PREF_SIZE    64
-#define MAX_PATH_COMPONENTS   20
-#define MAX_BIN_CHUNKS        110
-#else
-// Otherwise, the sky is the limit (complexity limit of 1 million verified instructions)
-#define MAX_STR_ARR_ELEM      128
-#define MAX_ARGS_STR_ARR_ELEM 128
-#define MAX_PATH_PREF_SIZE    128
-#define MAX_PATH_COMPONENTS   48
-#define MAX_BIN_CHUNKS        256
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 2, 0)  // lower values in old kernels (instr lim is 4096)
+#define MAX_STR_ARR_ELEM                40
+#define MAX_ARGS_STR_ARR_ELEM           15
+#define MAX_PATH_PREF_SIZE              64
+#define MAX_PATH_COMPONENTS             20
+#define MAX_BIN_CHUNKS                  110
+#else                                             // complexity limit of 1M verified instructions
+#define MAX_STR_ARR_ELEM                128
+#define MAX_ARGS_STR_ARR_ELEM           128
+#define MAX_PATH_PREF_SIZE              128
+#define MAX_PATH_COMPONENTS             48
+#define MAX_BIN_CHUNKS                  256
+#endif
+#else // CORE
+#define MAX_STR_ARR_ELEM                40        // TODO: turn this into global variables set w/ libbpfgo
+#define MAX_ARGS_STR_ARR_ELEM           15
+#define MAX_PATH_PREF_SIZE              64
+#define MAX_PATH_COMPONENTS             20
+#define MAX_BIN_CHUNKS                  110
+#endif
+
+/*================================ eBPF KCONFIGs =============================*/
+
+#if defined(bpf_target_x86)
+#define PT_REGS_PARM6(ctx)  ((ctx)->r9)
+#elif defined(bpf_target_arm64)
+#define PT_REGS_PARM6(x) (((PT_REGS_ARM64 *)(x))->regs[5])
```

Comment:
Why is this in the eBPF Kconfig section?

Reply:
this was copied and defined twice. I'm removing from this session in a new PR.

Reply:
https://github.com/aquasecurity/tracee/pull/1185